src/Squader.Api/Areas/Users/Controllers/UsersController.cs

```diff
@@ -34,5 +34,17 @@ public IActionResult GetUserById(Guid id)
             await this.commandBus.ExecuteAsync(command);
             return Ok();
         }
+
+        [HttpPost("CheckUsernameAvailability")]
+        public async Task<IActionResult> CheckUsernameAvailability([FromBody]CheckUsernameAvailabilityRequest request)
+        {
+            var query = new GetUserByIdentifiersQuery(request.Username);
+            var result = await Task.Run(() => this.queryBus.Execute(query));
+
+            if (result != null)
+                return StatusCode(409);
+
+            return StatusCode(200);          
```

Comment:
to nie jest błąd, ale zamiast StatusCode(200) możesz zwracać Ok() 

wiele http status code jest tak zaimplementowanych i pewnie ułatwiłoby to przeglądanie kodu osobie, która nazw tych kodów nie zna 

przykładowo zamiast 404 możesz zwrócić NotFound(), zamiast 201 Created (z parametrem), a zamiast 401 Unauthorized()